packages/react-components/src/components/chart/hooks/useDataZoom.ts

```diff
@@ -0,0 +1,64 @@
+import { MutableRefObject, useEffect, useState } from 'react';
+import { useViewport } from '../../../hooks/useViewport';
+import { EChartsType } from 'echarts';
+import { ECHARTS_GESTURE } from '../../../common/constants';
+import { DEFAULT_DATA_ZOOM } from '../eChartsConstants';
+import { ViewportInMs } from '../types';
+import { Viewport } from '@iot-app-kit/core';
+
+// known bug with zooming into live data https://github.com/apache/echarts/issues/11679
+// this function will prevent the data from moving if we are zoomed in
+const onDataZoomEvent = (chart: EChartsType, setViewport: (viewport: Viewport, lastUpdatedBy?: string) => void) => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  const currentDataZoom = chart.getOption().dataZoom[0];
+
+  const newDataZoom = {};
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  currentDataZoom.start > 0 ? (newDataZoom.startValue = currentDataZoom.startValue) : (newDataZoom.start = 0);
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  currentDataZoom.end < 100 ? (newDataZoom.endValue = currentDataZoom.endValue) : (newDataZoom.end = 100);
+
+  setViewport(
+    { start: new Date(currentDataZoom.startValue), end: new Date(currentDataZoom.endValue) },
+    ECHARTS_GESTURE
+  );
+  chart.setOption(
+    {
+      dataZoom: newDataZoom,
+    },
+    { lazyUpdate: true }
+  );
+};
+
+export const useDataZoom = (chartRef: MutableRefObject<EChartsType | null>, viewportInMs: ViewportInMs) => {
+  const { lastUpdatedBy, setViewport } = useViewport();
+  const [isScrolling, setIsScroling] = useState(false);
+
+  // handle live mode
+  useEffect(() => {
```

Comment:
do we need useEffect here ? given the if condition is gating the update anyway

Reply:
the useEffect is so that the dataZoom is only updated if the `viewportInMs` value changes

Reply:
but this will get triggered when either  isScrolling and or chartRef has changed as well right ? when you remove the chart.setoptions, and datazoom is a state,, you will be setting state here... it is not the best to NOT setState inside of a useEffect, https://www.youtube.com/watch?v=bGzanfKVFeU